src/java/main/org/apache/zookeeper/server/persistence/FileTxnSnapLog.java

```diff
@@ -165,8 +165,22 @@ public File getSnapDir() {
      */
     public long restore(DataTree dt, Map<Long, Integer> sessions,
             PlayBackListener listener) throws IOException {
-        snapLog.deserialize(dt, sessions);
+        long deserializeResult = snapLog.deserialize(dt, sessions);
         FileTxnLog txnLog = new FileTxnLog(dataDir);
+        if (-1L == deserializeResult) {
+            /* this means that we couldn't find any snapshot, so we need to
+             * initialize an empty database */
+            if (txnLog.getLastLoggedZxid() != -1) {
+                throw new IOException(
+                        "No snapshot found, but there are log entries. " +
+                        "Something is broken!");
+            }
+            /* TODO: (br33d) we should either put a ConcurrentHashMap on restore()
+             *       or use Map on save() */
+            save(dt, (ConcurrentHashMap<Long, Integer>)sessions);
```

Comment:
yes, ZOOKEEPER-261 is still a problem. is that what you are referring to? brian has a patch coming that builds on this one to fix that problem.